cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsNetworkConnector.java

```diff
@@ -135,18 +142,25 @@ public String getNetworkCidr(Network network, CloudCredential credential) {
     }
 
     @Override
-    public SubnetSelectionResult selectSubnets(List<CloudSubnet> subnetMetas, SubnetSelectionParameters subnetSelectionParameters) {
-        boolean preferPrivate = subnetSelectionParameters.isForDatabase() || subnetSelectionParameters.getTunnel().useCcm();
+    public SubnetSelectionResult filterSubnets(Collection<CloudSubnet> subnetMetas, SubnetSelectionParameters subnetSelectionParameters) {
+        boolean preferPrivate = subnetSelectionParameters.isPreferPrivateIfExist() || subnetSelectionParameters.getTunnel().useCcm();
 
-        SubnetSelectorStrategyType subnetSelectorStrategyType;
-        if (subnetSelectionParameters.isHa()) {
-            subnetSelectorStrategyType = preferPrivate ? SubnetSelectorStrategyType.MULTIPLE_PREFER_PRIVATE : SubnetSelectorStrategyType.MULTIPLE_PREFER_PUBLIC;
-        } else {
-            subnetSelectorStrategyType = preferPrivate ? SubnetSelectorStrategyType.SINGLE_PREFER_PRIVATE : SubnetSelectorStrategyType.SINGLE_PREFER_PUBLIC;
-        }
+        SubnetSelectorStrategyType subnetSelectorStrategyType = preferPrivate ?
```

Comment:
SubnetSelectorStrategyType could be renamed to filterStrategy as it does not select anymore.